StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1210UnitTests.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.IO;
+using System.Threading;";
+
+            await this.VerifyCSharpDiagnosticAsync(compilationUnit, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestProperOrderedUsingDirectivesInNamespaceDeclarationAsync()
+        {
+            var namespaceDeclaration = @"namespace Foo
+{
+    using System;
+    using System.Threading;
+}
+
+namespace Bar
+{
+    using System;
+    using Foo;
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(namespaceDeclaration, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInvalidOrderedUsingDirectivesInCompilationUnitAsync()
+        {
+            var compilationUnit = @"
+using System.Threading;
+using System.IO;
+using System;
+using System.Linq;";
+
+            DiagnosticResult[] expected = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(2, 1),
+                this.CSharpDiagnostic().WithLocation(3, 1)
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(compilationUnit, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInvalidOrderedUsingDirectivesInNamespaceDeclarationAsync()
+        {
+            var namespaceDeclaration = @"
+namespace Foo
+{
+    using System.Threading;
+    using System;
+}
+
+namespace Bar
+{
+    using Foo;
+    using Bar;
+    using System.Threading;
+    using System;
+}";
+
+            DiagnosticResult[] expected = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(4, 5),
+                this.CSharpDiagnostic().WithLocation(10, 5),
+                this.CSharpDiagnostic().WithLocation(12, 5)
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(namespaceDeclaration, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInvalidOrderedUsingDirectivesWithInlineCommentsAsync()
+        {
+            var namespaceDeclaration = @"
+namespace Foo
+{
+    using System;
+    using /*A*/ System.Threading;
+    using System.IO; //sth
+}";
+            DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(5, 5);
+
+            await this.VerifyCSharpDiagnosticAsync(namespaceDeclaration, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInvalidOrderedUsingDirectivesWithGlobalKeywordAsync()
+        {
+            var compilationUnit = @"
+using System.Threading;
+using global::System.IO;
+using global::System.Linq;
+using global::System;
+
+namespace Foo
+{
+    using global::Foo;
+    using System;
+}";
+
+            DiagnosticResult[] expected = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(2, 1),
+                this.CSharpDiagnostic().WithLocation(4, 1),
+            };
```

Comment:
I think the correct order would be:
```C#
using global::System;
using global::System.IO;
using global::System.Linq;
using System.Threading;
```
Basically that means ignoring the `global::` prefix which seems logical to me. 

If we would follow Visual Studio's way of ordering, diagnostics should be produced for both lines 3 and 4, as they both should be after the `global:System`.

@sharwell Do you know if there is a way to replace the built-in sort usings operator? If so, we could adapt the sort order, if not we should indeed seriously consider to follow Visual Studio's ordering.